src/main/java/core/stack/MateStack.java

```diff
@@ -1,19 +1,48 @@
 package core.stack;
 
+import java.util.EmptyStackException;
+
 public class MateStack<T> {
-    public void push(T value) {
+    private Node<T> top;
+    private int size;
 
+    public void push(T value) {
+        top = new Node<>(top, value);
+        size++;
     }
 
     public T peek() {
-        return null;
+        if (isEmpty()) {
+            throw new EmptyStackException();
+        }
+        return top.item;
     }
 
     public T pop() {
-        return null;
+        if (isEmpty()) {
+            throw new EmptyStackException();
+        }
+        T value = peek();
+        top = top.prev;
+        size--;
+        return value;
     }
 
     public int size() {
-        return 0;
+        return size;
+    }
```

Comment:
Adding a simple comment for the 'size' method would be helpful to indicate that it returns the number of elements in the stack.